File: backend/TaskManager/Entities/TaskItem.cs
using System.ComponentModel.DataAnnotations;
using TaskManager.Enums;

namespace TaskManager.Entities
{
    public class TaskItem
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Title { get; set; } = "";
        public string? Description { get; set; }
        public TaskItemStatus Status { get;set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: frontend/ng-task-manager/src/app/models/task-item.ts
export type TaskStatus = 0 | 1 | 2;

export interface TaskItem {
  id: string;
  title: string;
  description?: string;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
  isDeleted: boolean;
}

export const TaskStatusLabels = {
  0: 'Todo',
  1: 'InProgress',
  2: 'Done',
} as const;
